blockly-core/core/ui/block.js

```diff
@@ -1408,16 +1409,18 @@ Blockly.Block.prototype.setCurrentlyHidden = function (hidden) {
 };
 
 /**
- * Account for the fact that we have two different visibilty states.
+ * Account for the fact that we have two different visibility states.
  * UserVisible is a persisent property used to create blocks that can be seen
  * by level builders, but not by the user.
  * CurrentlyHidden is a non-persistent property used to hide certain blocks
  * (like function definitions/examples) that should only be visible when using
  * the modal function editor.
+ * This method calculates whether this block is currently visible
  * @returns true if both visibility conditions are met.
  */
 Blockly.Block.prototype.isVisible = function () {
-  return this.isUserVisible() && !this.isCurrentlyHidden();
+  return this.isUserVisible() && !this.isCurrentlyHidden_() &&
+    (!this.parentBlock_ || this.parentBlock_.isVisible());
```

Comment:
An alternative to this would be to make `isCurrentlyHidden_` a child-recursive and viral attribute in the same way that it looks like the serialized `userVisible` is (child blocks added get the parameter set on connection-time).